src/snapshot.rs

```diff
@@ -52,6 +53,26 @@ pub struct StartupData {
   raw_size: int,
 }
 
+impl From<Box<[u8]>> for StartupData {
+  fn from(slice: Box<[u8]>) -> Self {
+    let mut data: Vec<u8> = vec![];
+    data.copy_from_slice(&slice);
+    StartupData {
+      data: data.as_ptr() as *const char,
```

Comment:
When this function returns, `data: Vec<u8>` is dropped and `StartupData::data` points at uninitialized memory.

Reply:
ugh true. any idea what could be done instead for box?